etl/steps/data/garden/owid/latest/key_indicators/table_population.py

```diff
@@ -33,32 +36,116 @@ def make_table() -> Table:
         .pipe(calculate_aggregates)
         .pipe(prepare_dataset)
     )
+    _validate(t)
     t.metadata.short_name = "population"
     return t
 
 
+def _compare(a: pd.DataFrame, b: pd.DataFrame) -> pd.DataFrame:
+    assert a.index.names == b.index.names
+    index_names = a.index.names
+    a = a.reset_index()
+    b = b.reset_index()
+
+    a_keys = set(a[index_names].itertuples(index=False, name=None))
+    b_keys = set(b[index_names].itertuples(index=False, name=None))
+
+    if a_keys - b_keys:
+        raise AssertionError(f"{a_keys - b_keys}")
+    if b_keys - a_keys:
+        raise AssertionError(f"{b_keys - a_keys}")
+
+    return a.compare(b)
+
+
+def _validate(t: Table) -> None:
+    """Is the new population data identical to a version from importers?"""
+    COLS_MAP = {
+        "Entity": "country",
+        "Year": "year",
+        "Population (historical estimates and future projections)": "population",
+        "Share of world population": "world_pop_share",
+    }
+    # transform it into new shape
+    old_df = (
+        pd.read_csv(DIR_PATH / "output" / "Population (Gapminder, HYDE & UN).csv")
+        .rename(columns=COLS_MAP)[COLS_MAP.values()]
+        .set_index(["country", "year"])
+    )
+
+    # TODO: South America is not yet ready, fix this after https://github.com/owid/etl/issues/92
```

Comment:
South America & Latina America is blocked by https://github.com/owid/etl/issues/92.